pkg/collector/corechecks/servicediscovery/service_detector.go

```diff
@@ -6,6 +6,9 @@
 package servicediscovery
 
 import (
+	"golang.org/x/exp/slices"
```

Comment:
Could we use the `slices` package instead? I've seen problems in other big mono-repos with packages from `golang.org/x/exp`, since its an experimental package they can introduce breaking changes in any commit.